data_pipelines/assets/flood/discharge_new.py

```diff
@@ -0,0 +1,231 @@
+import os
+from datetime import datetime, timedelta
+import pandas as pd
+
+import xarray as xr
+from dagster import AssetExecutionContext, asset
+import dask.dataframe as dd
+
+from data_pipelines.resources.dask_resource import DaskResource
+from data_pipelines.resources.glofas_resource import CDSClient
+from data_pipelines.utils.flood.config import *
+from data_pipelines.utils.flood.etl.filter_by_upstream import apply_upstream_threshold
+from data_pipelines.utils.flood.etl.raster_converter import RasterConverter
+from data_pipelines.utils.flood.etl.transforms import (
+    compute_flood_intensity,
+    compute_flood_peak_timing,
+    compute_flood_tendency,
+    compute_flood_threshold_percentages,
+    add_geometry,
+)
+from data_pipelines.utils.flood.etl.utils import restrict_dataset_area
+
+from data_pipelines.partitions import discharge_partitions
+
+
+def make_path(*args) -> str:
+    path = os.path.join(*args)
+    os.makedirs(os.path.dirname(path), exist_ok=True)
+    return path
+
+
+# This is the partitioned version of the raw_discharge asset
+# It performs prallelized queries of the CDS API for each leadtime hour
+# Such parallelization can cause the pipeline to fail with no clear error message
+# Using a tag for limiting concurrency can help prevent this
+@asset(
+    key_prefix=["flood"],
+    partitions_def=discharge_partitions,
+    io_manager_key="grib_io_manager",
+)
+def raw_discharge(context: AssetExecutionContext, client: CDSClient) -> None:
+    date_for_request = datetime.utcnow() - timedelta(days=TIMEDELTA)
+
+    query_buffer = GLOFAS_RESOLUTION * GLOFAS_BUFFER_MULT
+    lat_min = GLOFAS_ROI_CENTRAL_AFRICA["lat_min"]
+    lat_max = GLOFAS_ROI_CENTRAL_AFRICA["lat_max"]
+    lon_min = GLOFAS_ROI_CENTRAL_AFRICA["lon_min"]
+    lon_max = GLOFAS_ROI_CENTRAL_AFRICA["lon_max"]
+
+    area = [
+        lat_max + query_buffer,
+        lon_min - query_buffer,
+        lat_min - query_buffer,
+        lon_max + query_buffer,
+    ]
+
+    if USE_CONTROL_MEMBER_IN_ENSEMBLE:
+        product_type = ["control_forecast", "ensemble_perturbed_forecasts"]
+        print("Retrieving both control and ensemble")
+    else:
+        product_type = "ensemble_perturbed_forecasts"
+        print("Retrieving only ensemble")
+
+    l_hour = context.partition_key
```

Comment:
Could this be renamed to f.ex. `leadtime_hour`? I confused the `l` with `1` at first.